test/CLAUDE.md

```diff
@@ -0,0 +1,103 @@
+- **`test_export.py`** - Model export functionality
+- **`test_quantization.py`** - Quantization algorithms
+- **`test_distributed.py`** - Multi-process training
+- **`test_profiler.py`** - Performance profiling tools
+- **`test_jit.py`** - TorchScript compilation and execution (note: TorchScript is deprecated)
+
+#### Backend-Specific Directories
+- **`autograd/`** - Autograd-specific test cases
+- **`backends/`** - Backend implementations testing
+- **`ao/`** - AO (Accelerated Operations) quantization tests
+
+### Supporting Infrastructure
+- **`benchmark_utils/`** - Performance benchmarking utilities
+- **`bottleneck_test/`** - Performance bottleneck identification
+- **`compiled_autograd_skips/`** - Known compilation issues tracking
+
+## 🧪 Running Tests
+
+### Basic Test Execution
+```bash
+# Ask the user for which test should run
+
+# Run specific test file
+python test/run_test.py -i test_torch
+python test/run_test.py -i test_nn
+python test/run_test.py -i test_autograd
+
+# Run tests with specific patterns
+python test/run_test.py -k "test_add"
+python test/run_test.py -k "TestNN"
+python test/run_test.py test_torch.py::TestTorch::test_add_cuda
+```
+
+## 🔧 Test Development Patterns
+
+### Identifying Relevant Tests
+Part of development work is figuring out which tests are relevant for your changes:
+
+- **Operator changes**: Look at `test_ops.py`, `test_torch.py`
+- **Autograd changes**: Check `test_autograd.py`, `autograd/` directory
+- **NN module changes**: Run `test_nn.py`, specific module tests
+- **Backend changes**: Test device-specific files (`test_cuda.py`, etc.)
+- **Build system changes**: Test core functionality and CI integration
+
+### Writing New Tests
+
+Test organization principles:
+- Add new test class only for major new features
+- Add new test methods for unit tests within existing classes
+- For new operators, add OpInfo entries rather than individual test methods
+
+### Common Test Utilities
+- **`torch.testing.assert_close()`** - Tensor comparison with tolerances
+- **`@parametrize`** - Test multiple configurations
+- **`@skipIf`** - Conditional test skipping
+- **`gradcheck`** - Automatic gradient verification
+
+### Test Instantiation Patterns
+- `instantiate_device_type_tests()` - Generates device-specific test variants (CPU, CUDA, etc.)
+- `instantiate_parametrized_tests()` - Creates parameterized test combinations
+- Test methods with extra args (device, dtype) are automatically called from these instantiated new tests
+
+## 📝 Notes for Claude
+
+- Test discovery is based on filename patterns (`test_*.py`)
+- Tests are designed to be run independently and in parallel
+- Device-specific tests automatically skip if hardware unavailable
+- Heavy use of parameterization for testing multiple configurations
+- CI runs tests in sharded mode across multiple workers
+- Test results are often XML-formatted for CI integration
+- Memory and performance tests require special attention to cleanup
```

Comment:
Some of these are kinda helpful, but this seems like one that isn't - if we don't give it a more specific bit of guidance I am not sure this is adding much.